### PR TITLE
feat(FX-4797): Automatically persist any pending artwork lists changes in sheet

### PR DESCRIPTION
**Pull request by dimatretyak**
This PR resolves [FX-4797] 

### Demo
| Android | iOS |
| ------------- | ------------- |
| <video src="https://github.com/artsy/eigen/assets/3513494/e8f7266d-b36e-4241-9914-7fba44b96185" /> | <video src="https://github.com/artsy/eigen/assets/3513494/cb71b158-ed2b-4de0-aa16-6d85feed1161" /> | 

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [ ] I added an **[app state migration]**, or my changes do not require one.
- [ ] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- Automatically persist any pending artwork lists changes in sheet - dimatretyak

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[FX-4797]: https://artsyproduct.atlassian.net/browse/FX-4797?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/Components/ArtworkLists/views/SelectArtworkListsForArtworkView/SelectArtworkListsForArtworkView.tsx [hunk 1]
```diff
@@ -6,20 +6,32 @@ import { useArtworkListsBottomOffset } from "app/Components/ArtworkLists/useArtw
...
   const bottomOffset = useArtworkListsBottomOffset(2)
+  const { save } = useSavePendingArtworkListsChanges()
+  const hasUnsavedChanges = state.hasUnsavedChanges
+
+  const onDismiss = useCallback(() => {
+    if (hasUnsavedChanges) {
+      save()
+    }
+
+    reset()
```
Comment (dimatretyak) [inline src/app/Components/ArtworkLists/views/SelectArtworkListsForArtworkView/SelectArtworkListsForArtworkView.tsx:26]: we have to check this code carefully on race condition (especially with bad internet connection and for android) ☝️

Comment (dimatretyak) [reply src/app/Components/ArtworkLists/views/SelectArtworkListsForArtworkView/SelectArtworkListsForArtworkView.tsx:26]: Beta builds with changes:
* Android - 8.14.0 (1674645319)
* iOS - 8.14.0 in review and therefore we [can't build](https://app.circleci.com/pipelines/github/artsy/eigen/42613/workflows/36fa5be5-1bad-40c6-8c32-a43bfe2ac164/jobs/139319) a build

**Review by dblandin — APPROVED** [review]
LGTM! Hopefully we can reduce the delay between dismissal and confirmation toast as well. I know we're tracking that investigation in [another ticket](https://artsyproduct.atlassian.net/browse/FX-4798).

**Review by anandaroop — APPROVED** [review]
Looks good 👍🏽

**Pull request merged**